services/provider/server/server.go

```diff
@@ -799,12 +794,25 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 		return nil, status.Errorf(codes.Internal, "Failed to update lastHeartbeat payload in the storageConsumer resource: %v", err)
 	}
 
+	storageConsumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to get storageConsumer resource: %v", err)
+	}
+
 	channelName, err := s.getOCSSubscriptionChannel(ctx)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Failed to construct status response: %v", err)
 	}
+	var arr = make([]string, 0)
+	arr = append(arr, "DesiredClientOperatorChannel", channelName)
+	if storageConsumer.Spec.StorageQuotaInGiB > 0 {
+		clusterResourceQuotaSpec := getClusterResourceQuotaSpec(storageConsumer)
+		arr = append(arr, "ClusterResourceQuotaSpec", fmt.Sprintf("%v", clusterResourceQuotaSpec))
+	}
```

Comment:
1. You are hashing too much, there is no reason to hash the entire spec. You need only the value for the quota which comes directly from the storage consumer spec
2. You should hash the quota even if it is not set. The data to create the ash should be predictable and not conditional